oculus_sonar:
  ros__parameters:
    frame_id: "sonar" # Frame of reference for the ping messages. Default value is "sonar".

    temperature_warn: 30. # Sonar temperature at which a warning is raised. Default value is 30.0.
    temperature_stop: 35. # Sonar temperature at which the sonar goes to standby mode. Default value is 35.0.

    run: False # If run is False, stanby mode is forced. Default value is False.

    frequency_mode: 1 # Sonar beam frequency mode. Default value is 1.
    # 1: Low frequency (1.2MHz, wide aperture).
    # 2: High frequency (2.1MHz, narrow aperture).

    ping_rate: 0 # Frequency of ping fires. Default value is 2.
    # 0: 10Hz max ping rate.
    # 1: 15Hz max ping rate.
    # 2: 40Hz max ping rate.
    # 3: 5Hz max ping rate.
    # 4: 2Hz max ping rate.
    # 5: Standby mode (no ping fire).

    data_depth: 1 # Ping data encoding bit count. Default value is 1.
    # 0: Ping data encoded on 8bits.
    # 1: Ping data encoded on 16bits.

    nbeams: 1 # Number of ping beams. Default value is 1.
    #  0: Oculus outputs 256 beams.
    #  1: Oculus outputs 512 beams.

    gain_assist: True # Enable gain assist. Default value is True.

    range: 20.0 # Sonar range (in meters), min=0.3, max=40.0.  Default value is 20.0.

    # These parameters are for scaling data (investigate)
    gamma_correction: 158 # Gamma correction, min=0, max=255.  Default value is 158.
    gain_percent: 50.0 # Gain percentage (%), min=0.1, max=100.0.  Default value is 50.0.

    sound_speed: 0.0 # Sound speed (in m/s, set to 0 for it to be calculated using salinity), min=1400.0, max=1600.0. Default value is 0.0.
    use_salinity: True # Use salinity to calculate sound_speed. Default value is True.
    salinity: 0.0 # Salinity (in parts per thousand (ppt,ppm,g/kg), used to calculate sound speed if needed), min=0.0, max=100. Default value is 0.0.

